### PR TITLE
Add event for the calculation of town map color

### PR DESCRIPTION
**Pull request by Goosius1**
#### Description: 
- I need an event for something I'm doing at the moment.
- This event will be when a town returns its map colour.
- This is to allow siegewar (for example) to intervene in the display of individual town map (ie dynmap) colours.

____
#### New Nodes/Commands/ConfigOptions: 
N/A

____
#### Relevant Towny Issue ticket:
N/A

____
- [N/A] I have tested this pull request for defects on a server. 

By making this pull request, I represent that I have the right to waive copyright and related rights to my contribution, and agree that all copyright and related rights in my contributions are waived, and I acknowledge that the TownyAdvanced organization has the copyright to use and modify my contribution under the Towny [License](https://github.com/LlmDl/Towny/blob/master/LICENSE.md) for perpetuity.

### CODE REVIEW

**Review by LlmDl — APPROVED** [review]
And I'm guessing that the getMapColorHexCode() is being called by something like SiegeWar and not internally?

**Comment by Goosius1** [comment]
- That's right. It's called by the Dynmap-Towny plugin

File: src/com/palmergames/bukkit/towny/object/Town.java [hunk 1]
```diff
@@ -1328,6 +1330,13 @@ public long getRuinedTime() {
 		return ruinedTime;
 	}
 
+	public String getMapColorHexCode() {
+		String rawMapColorHexCode = hasNation() ? nation.getMapColorHexCode() : null;
```
Comment (silverwolfg11) [inline src/com/palmergames/bukkit/towny/object/Town.java:1334]: Seems like this method can potentially return `null`. The method should probably be annotated with a `@Nullable` then.

File: src/com/palmergames/bukkit/towny/event/town/TownMapColourCalculationEvent.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+		super(!Bukkit.getServer().isPrimaryThread());
+		this.town = town;
+		this.mapColorHexCode = mapColorHexCode;
+	}
+
+	public Town getTown() {
+		return town;
+	}
+
+	public String getMapColorHexCode() {
```
Comment (silverwolfg11) [inline src/com/palmergames/bukkit/towny/event/town/TownMapColourCalculationEvent.java:27]: If I'm reading this right `mapColorHexCode` can be `null` so the method should probably be annotated `@Nullable` just to let other devs know that.

**Review by silverwolfg11 — COMMENTED** [review]
Looks good! I think a few nullability annotations might be helpful though.

File: src/com/palmergames/bukkit/towny/object/Town.java (hunk 1 above)
Comment (Goosius1) [reply src/com/palmergames/bukkit/towny/object/Town.java:1335]: Makes sense, done.

File: src/com/palmergames/bukkit/towny/event/town/TownMapColourCalculationEvent.java (hunk 2 above)
Comment (Goosius1) [reply src/com/palmergames/bukkit/towny/event/town/TownMapColourCalculationEvent.java:29]: Makes sense, done.

**Review by LlmDl — APPROVED** [review]

**Pull request merged**